test/merge-with-rules.test.ts

```diff
@@ -496,4 +496,45 @@ describe("Merge with rules", function () {
 
     expect(() => _mergeWithoutRule(config, config)).not.toThrow();
   });
+
+  it("should merge with append only (#151)", function () {
```

Comment:
I think test that tests both behaviors (`match` rules and `only` rules) in a single merge is a must.

Reply:
Yup, I need to add some mixed test(s). Likely your use case is a nice one to include.